secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/sign/SignPayloadUtil.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.sign;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.forgerock.json.jose.builders.JwtBuilderFactory;
+import org.forgerock.json.jose.builders.SignedJwtBuilderImpl;
+import org.forgerock.json.jose.jws.JwsAlgorithm;
+import org.forgerock.json.jose.jws.SignedJwt;
+import org.forgerock.json.jose.jws.SigningManager;
+import org.forgerock.json.jose.jws.handlers.SigningHandler;
+import org.forgerock.json.jose.jwt.JwtClaimsSet;
+import org.forgerock.secrets.NoSuchSecretException;
+import org.forgerock.secrets.Purpose;
+import org.forgerock.secrets.SecretsProvider;
+import org.forgerock.secrets.keys.SigningKey;
+import org.forgerock.util.Reject;
+import org.forgerock.util.promise.Promise;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Util to sign objects <br/>
+ * Implementation of {@link SignUtil}
+ */
+public class SignPayloadUtil implements SignUtil {
```

Comment:
I like that you are using an interface but I think it is currently too generic, it is better to make it more specific to begin with then make it more generic if we find other cases where it can be reused.

I would name it something like this:
- ObieJwsSigner (interface)
- DefaultObieJwsSigner (class)

We get the flexibility of using an interface, allowing other impls to be used in future but we don't have to invest lots of time now making it really generic to be used for any Signing use case (as this will take a lot of design thought).

The current logic is OB specific at the moment and produces Jws sigs only, so I want that captured in the name.

Reply:
Renamed and added few improvements to be more generic so that the customer can implement his own signers.